fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -348,6 +353,10 @@ def set_gradient_divide_factors(self, pre: float, post: float, recursive: bool)
         self.gradient_predivide_factor = pre
         self.gradient_postdivide_factor = post
 
+    @property
```

Comment:
how about we move all deprecated code to the end of the class?

Reply:
Done.